sso/resources/org.jaggeryjs.modules.sso/src/main/java/org/jaggeryjs/modules/sso/common/util/Util.java

```diff
@@ -192,6 +202,53 @@ public static XMLObject unmarshall(String authReqStr) throws Exception {
         }
     }
 
+    /**
+     * Return whether SAML Assertion has the canonicalization method
+     * set to 'http://www.w3.org/2001/10/xml-exc-c14n#WithComments'.
+     *
+     * @param document
+     * @return true if canonicalization method equals to 'http://www.w3.org/2001/10/xml-exc-c14n#WithComments'
+     */
+    private static boolean isSignedWithComments(Document document) {
+
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        try {
+            String assertionId = (String) xPath.compile("//*[local-name()='Assertion']/@ID")
+                    .evaluate(document, XPathConstants.STRING);
+
+            if (StringUtils.isBlank(assertionId)) {
+                return false;
+            }
+
+            NodeList nodeList = ((NodeList) xPath.compile(
+                    "//*[local-name()='Assertion']" +
```

Comment:
shouldn't we use shipbuilder?

Reply:
Since this is are static strings.. compiler will handle it.